### ISSUE TITLE
Projection Pursuit Regression and Classification

### ISSUE THREAD

**Issue by pavelkomarov** [scikit-learn-contrib/scikit-learn-contrib#30]
Request for project inclusion in scikit-learn-contrib

Hi guys, I've created a thing I think should belong to scikit-learn-contrib so more people can know about it. I've gotten mostly through the workflow (still trying to figure out exactly how to run sphinx and make my documentation pretty and launch it automatically with circleci) and would like some feedback as to whether you want this and, if you feel it is not ready, what more I need to do before you're willing to accept it.

* Project name: projection-pursuit
* Project description: An implementation of Jerome Friedman's multivariate projection pursuit regression algorithm and classification extension.
* Authors: Me
* Current repository: https://github.com/pavelkomarov/projection-pursuit
* Requirements:
 * [yes] scikit-learn compatible (check_estimator passed)
 * [ish] Documentation (guide, API reference, example gallery)
 * [85] Unit tests (coverage: xx%)
 * [yes] Python3 compatible
 * [afaik] PEP8 compliant
 * [yes] Continuous integration

Regards,

**Comment by pavelkomarov** [comment]
Last night and today I took a foray in to doing documentation with continuous integration. I am using TravisCI for this purpose, because CircleCI is over-complicated for what I need. It works but not completely:

* Adding `'sphinx_gallery.gen_gallery'` to the `extensions` in `conf.py` for some reason removes the `_modules` folder from the output, so all the `[source]` links in the documentation break.
* `.rst`s autogenerated by `gen_gallery` are causing undefined label warnings.
* The gallery `index.html` is full of examples that are not clickable links.
* I've added `'travis_button': True, 'codecov_button': True ` and a few other things to `html_theme_options` in `conf.py`, but buttons aren't showing up as the Alabaster documentation says they should.
* It's not clear to me where I should put the toctree full of plot examples and a reference to `skpp.rst`.
* There are still stupid details like "This Page" in the top left that I'm not sure how to get rid of.

https://rawgit.com/pavelkomarov/projection-pursuit/gh-pages/skpp.html. I've been at it a while and am kind of annoyed with Sphinx. If any of you are a Sphinx wiz and know exactly how to fix some of these let me know.

**Comment by anbarief** [comment]
Projection Pursuit is a very special case (simplified case with fewer parameters) of Neural Network model.

**Comment by pavelkomarov** [comment]
It can be thought of that way, but the method of training is very different.

**Comment by pavelkomarov** [comment]
I would argue it's different enough it shouldn't just be outright ignored https://www.econstor.eu/bitstream/10419/61280/1/721436943.pdf. There is compression to low-dimensional spaces where a low-dimensional function is fit, somewhat different from what a neural net is typically doing. It's also a neat model to see [the relationship between parameters and output to get a sense of the training process in ML](pavelkomarov.com/projection-pursuit/math.pdf).
